python/cudf/dataframe/series.py

```diff
@@ -857,6 +857,11 @@ def sum(self, axis=None, skipna=True):
         assert axis in (None, 0) and skipna is True
         return self._column.sum()
 
+    def product(self, axis=None, skipna=True):
```

Comment:
Is this a new API addition that was missing before? If so, it might be good to add a line under new features in CHANGELOG? @kkraus14 what do you think?

Reply:
Agreed. Completely slipped my mind since `product` isn't a commonly used reduction 😄

Reply:
CHANGELOG is updated accordingly.